File: backend/src/main/java/com/wolyh/game/backend/dto/RoomResponses.java
package com.wolyh.game.backend.dto;

import java.util.List;
import java.util.Set;

import com.wolyh.game.backend.model.UnitCoordinates;

public class RoomResponses {
    public static record CreateRoom(String roomId) {}
    public static record JoinRoom(String creatorName, String roomId) {}
    public static record StartGame(
        Set<String> fov, 
        List<UnitCoordinates> unitSpawns,
        int nb_units,
        String roomId
    ) {}
}

File: frontend/src/dto/RoomResponses.ts
export namespace RoomResponses {
    export interface CreateRoom {
        roomId: string;
    }

    export interface JoinRoom {
        creatorName: string;
        roomId: string;
    }

    export interface StartGame {
        fov: Array<string>;
        unitSpawns: Array<{idx: number, q: number, r: number}>;
        nb_units: number;
        roomId: string;
    }
}